test/extensions/filters/common/local_ratelimit/local_ratelimit_test.cc

```diff
@@ -386,27 +397,23 @@ TEST_F(LocalRateLimiterDescriptorImplTest, TokenBucketMultipleTokensPerFillDescr
   EXPECT_FALSE(rate_limiter_->requestAllowed(descriptor_));
 
   // 0 -> 2 tokens
-  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
-                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
-  EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
-  fill_timer_->invokeCallback();
-  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
-                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
-  EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
-  fill_timer_->invokeCallback();
+  for (int i = 0; i < 2; i++) {
+    dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
+                                                     Envoy::Event::Dispatcher::RunType::NonBlock);
+    EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
+    fill_timer_->invokeCallback();
+  }
 
   // 2 -> 1 tokens
   EXPECT_TRUE(rate_limiter_->requestAllowed(descriptor_));
 
   // 1 -> 2 tokens
-  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
-                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
-  EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
-  fill_timer_->invokeCallback();
-  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
-                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
-  EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
-  fill_timer_->invokeCallback();
+  for (int i = 0; i < 2; i++) {
+    dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
+                                                     Envoy::Event::Dispatcher::RunType::NonBlock);
+    EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
+    fill_timer_->invokeCallback();
+  }
```

Comment:
Sorry, but I did't get why you make these change?

Reply:
The existing tests utilized two different ways to implement it: 

1. [Duplicating the calls](https://github.com/envoyproxy/envoy/blob/9d9940b49f642fcf483ac8ff387ecb521dcf94fc/test/extensions/filters/common/local_ratelimit/local_ratelimit_test.cc#L388):
```c++
  // 0 -> 2 tokens
  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
  EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
  fill_timer_->invokeCallback();
  dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
                                                   Envoy::Event::Dispatcher::RunType::NonBlock);
  EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
  fill_timer_->invokeCallback();
```

2. [Using for loops](https://github.com/envoyproxy/envoy/blob/9d9940b49f642fcf483ac8ff387ecb521dcf94fc/test/extensions/filters/common/local_ratelimit/local_ratelimit_test.cc#L545):
```c++
  // 0 -> 2 tokens for descriptor_
  for (int i = 0; i < 60; i++) {
    dispatcher_.globalTimeSystem().advanceTimeAndRun(std::chrono::milliseconds(50), dispatcher_,
                                                     Envoy::Event::Dispatcher::RunType::NonBlock);
    EXPECT_CALL(*fill_timer_, enableTimer(std::chrono::milliseconds(50), nullptr));
    fill_timer_->invokeCallback();
  }
```

Since I had to update the tests to validate the changes that were implemented, I took the opportunity to standardize how it is done.

In my opinion the second format is clearer and easier to understand than the first one.